tests/std/tests/LWG2381_num_get_floating_point/test.cpp

```diff
@@ -0,0 +1,509 @@
+};
+
+class mid_zero_numpunct : public numpunct<char> {
+public:
+    mid_zero_numpunct() : numpunct<char>() {}
+
+protected:
+    virtual string do_grouping() const {
+        return "\1\0\2"s;
+    }
+};
+
+class my_numpunct : public numpunct<char> {
+public:
+    my_numpunct() : numpunct<char>() {}
+
+protected:
+    virtual char_type do_decimal_point() const {
+        return ';';
+    }
+    virtual char_type do_thousands_sep() const {
+        return '_';
+    }
+    virtual string do_grouping() const {
+        return string("\1\2\3");
+    }
+};
+
+template <class Flt>
+void test() {
+    const my_facet f(1);
+    ios instr(nullptr);
+    Flt v = 0;
+    {
+        const char str[] = "123";
+        assert((instr.flags() & instr.basefield) == instr.dec);
+        assert(instr.getloc().name() == "C");
+        ios_base::iostate err = instr.goodbit;
+        const char* iter      = f.get(str, str + sizeof(str), instr, err, v);
+        assert(iter == str + sizeof(str) - 1);
+        assert(err == instr.goodbit);
+        assert(v == 123);
+    }
+    {
+        const char str[]      = "-123";
+        ios_base::iostate err = instr.goodbit;
+        const char* iter      = f.get(str, str + sizeof(str), instr, err, v);
+        assert(iter == str + sizeof(str) - 1);
+        assert(err == instr.goodbit);
+        assert(v == -123);
+    }
+    {
+        const char str[]      = "123.5";
+        ios_base::iostate err = instr.goodbit;
+        const char* iter      = f.get(str, str + sizeof(str), instr, err, v);
+        assert(iter == str + sizeof(str) - 1);
+        assert(err == instr.goodbit);
+        assert(v == 123.5);
+    }
+    {
+        const char str[] = "125e-1";
+        hex(instr);
+        ios_base::iostate err = instr.goodbit;
+        const char* iter      = f.get(str, str + sizeof(str), instr, err, v);
+        assert(iter == str + sizeof(str) - 1);
+        assert(err == instr.goodbit);
+        assert(v == 125e-1);
+    }
+    {
+        const char str[] = "0x125p-1";
+        hex(instr);
+        ios_base::iostate err = instr.goodbit;
+        const char* iter      = f.get(str, str + sizeof(str), instr, err, v);
+        assert(iter == str + sizeof(str) - 1);
+        assert(err == instr.goodbit);
+        assert(v == static_cast<double>(0x125) * 0.5);
+    }
+    {
+        v                     = -1;
+        const char str[]      = "123_456_78_9;125";
+        ios_base::iostate err = instr.goodbit;
+        const char* iter      = f.get(str, str + sizeof(str), instr, err, v);
+        assert(iter == str + 3);
+        assert(err == instr.goodbit);
+        assert(v == 123);
+    }
+    {
+        v                     = -1;
+        const char str[]      = "2-";
+        ios_base::iostate err = instr.goodbit;
+        const char* iter      = f.get(str, str + sizeof(str), instr, err, v);
+        assert(iter == str + 1);
+        assert(err == instr.goodbit);
+        assert(v == 2);
+    }
+

```

Comment:
There's a newline here, but not between the other blocks, seems like this should be removed.